src/AccordionSection/AccordionSection.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from "react"
+import { SectionHeader } from "../Internals/SectionHeader"
+import styled from "../utils/styled"
+import { ChevronUpIcon, ChevronDownIcon } from "../Icon/Icon"
+import { DefaultProps } from "../types"
+import isFunction from "lodash/isFunction"
+
+export interface AccordionSectionProps extends DefaultProps {
+  title: React.ReactNode
+  children: (() => React.ReactNode) | React.ReactNode
+  // for internal use
+  index?: number
```

Comment:
Maybe prefix with an `_`?

```suggestion
  _index?: number
```